src/Windows/Avalonia.Win32/Win32Platform.cs

```diff
@@ -19,20 +21,24 @@
 using Avalonia.Threading;
 using Avalonia.Win32.Input;
 using Avalonia.Win32.Interop;
+using Avalonia.Win32.Notifications;
 using static Avalonia.Win32.Interop.UnmanagedMethods;
 
 namespace Avalonia
 {
     public static class Win32ApplicationExtensions
     {
-        public static T UseWin32<T>(
-            this T builder) 
-                where T : AppBuilderBase<T>, new()
+        public static T UseWin32<T>(this T builder)
+            where T : AppBuilderBase<T>, new()
         {
-            return builder.UseWindowingSubsystem(
-                () => Win32.Win32Platform.Initialize(
-                    AvaloniaLocator.Current.GetService<Win32PlatformOptions>() ?? new Win32PlatformOptions()),
-                "Win32");
+            var settings = AvaloniaLocator.Current.GetService<Win32PlatformOptions>() ?? new Win32PlatformOptions();
+            return builder
+                .UseWindowingSubsystem(() => Win32.Win32Platform.Initialize(settings), "Win32")
+                .AfterSetup(b =>
+                {
+                    Win32.Win32Platform.SetupAppUserModelId(settings.AppUserModelId ?? b.Instance.Name);
```

Comment:
1) this should live in `Win32.Win32Platform.Initialize`
2) add a check for the windows version, we probably don't want to do that on Win7/Win8.1

Reply:
> this should live in Win32.Win32Platform.Initialize

I don't get the `Application` instance during `Win32.Win32Platform.Initialize` - it's `null`

Reply:
Then pass the AppBuilderBase instance to `Initialize` and call AfterSetup hooks there. `UseWindowingSubsystem` is called by previewer after `BuildAvaloniaApp` is already called and it overrides the windowing subsystem. So your platform-specific AfterSetup hook is called when it shouldn't be.

Reply:
Do we still need the parameterless `public static void Initialize()`?

Reply:
I don't think so, the only stable public contracts of platform backends are `Use[PlatformName]` and `[PlatformName]PlatformOptions`

Reply:
So should I just pass `null` as `AppBuilderBase<T>`? What is `T` supposed to be though? Also seems like a weird side-effect of calling this function that NotificationManager doesn't work.

Reply:
>So should I just pass null as AppBuilderBase<T>

You have `this T builder` right there

Reply:
Oh I didn't read it correctly; I though we still need the parameterless constructor and was confused what to pass as the builder.